data/strings/strings.txt

```diff
@@ -26313,6 +26313,62 @@
     ru = Если эта функция включена, вам не нужно каждый раз разблокировать устройство во время работы приложения.
     tr = Etkinleştirildiğinde, uygulama çalışırken her seferinde cihazınızın kilidini açmanıza gerek kalmaz.
     uk = Якщо ввімкнено, вам не потрібно щоразу розблоковувати пристрій під час роботи програми.
+  
+  [search_history]
+    tags = android
+    comment = Search History Privacy
+    en = Keep Search History On
+    af = Hou Soekgeskiedenis Aan
+    ar = الاحتفاظ بتاريخ البحث
+    be = Захаваць гісторыю пошуку
+    ca = Mantén l'historial de cerca
+    cs = Ponechte historii vyhledávání
+    de = Suchverlauf behalten
+    es = Mantener el historial de búsqueda
+    et = Säilita otsingu ajalugu
+    eu = Mantendu bilaketa historia
+    fi = Säilytä hakuhistoria
+    fr = Conserver l'historique de recherche
+    hu = Keresési előzmények megtartása
+    it = Mantieni la cronologia delle ricerche
+    lt = Išlaikyti paieškos istoriją
+    mr = शोध इतिहास कायम ठेवा
+    nb = Behold søkehistorikken
+    nl = Bewaar zoekgeschiedenis
+    pl = Zachowaj historię wyszukiwania
+    pt-BR = Manter histórico de pesquisa
+    ro = Păstrați istoricul căutărilor
+    ru = Сохранить историю поиска
+    tr = Arama Geçmişini Sakla
+    uk = Зберегти історію пошуку
```

Comment:
Not all translations are present. Did you use tools/python/translate.py?

Reply:
how can I use translate.py?
it's promptig error while running:

``````````
C:\Users\ASUS>python translate.py
Traceback (most recent call last):
  File "C:\Users\ASUS\translate.py", line 9, in <module>
    import requests
ModuleNotFoundError: No module named 'requests'

Reply:
Did you try `python -m pip install requests` ?

Reply:
after running the file it shows an error :

``````````````
D:\git\organicmaps\tools\python>python translate.py Search History is On
Error: neither DEEPL_FREE_API_KEY nor DEEPL_API_KEY environment variables are set.
DeepL translations are not available. Register for a free Developer API account here:
https://www.deepl.com/pro#developer
and get the API key here: https://www.deepl.com/account/summary

Reply:
@biodranik  can i add the rest of the languages manually ?

Reply:
Yes, of course. Was it hard to set up a DeepL account?

Reply:
I did set up the account but it says that it only accepts credits(even free ones) from very few countries.....so I couldnt get the DEEPL_API_KEY

Reply:
You may try `tools/unix/translate_categories.sh` instead.

Reply:
Shouldn't I add the last three languages manually ?

Reply:
tools/unix/translate_categories.sh  is showing syntax error again:

`````````````

D:\git\organicmaps\tools\unix>python translate_categories.sh
  File "D:\git\organicmaps\tools\unix\translate_categories.sh", line 19
    1) SRC=en
     ^
SyntaxError: unmatched ')'

Reply:
@biodranik   What is the purpose of translate_categories.sh ? WIll it translate and add all the languages automatically or just generate the same line in different languages so that we can paste them in the strings.xml files?

Reply:
Run it to see the help.

Reply:
I see, you got some errors. How exactly are you running it? Using WSL? Or git bash?

Reply:
I used WSL bash to run it 

``````````
bash translate_categories.sh  "Keep Search History On"


````````````
And it gave the output in many languages. But the thing is , I added them manually in all the strings.xml files